### PR TITLE
Add "Document Information Extraction with LLM" article.

### PR DESCRIPTION
**Pull request by Carvin618**
Reproduced chards and Images URL: https://www.figma.com/design/GHDhmnzRQTAUdjNTgOPJmF/57Blocks-Website-V3?node-id=14042-2730&t=yRErkUsA0v9kn6JO-0

### CODE REVIEW

**Comment by live9080** [comment]
Can you please rebase or merge the latest 'dev' branch? We have added an action in the 'dev' branch that automatically generates a preview URL. After you update your PR, you should be able to get the preview URL added by the bot(like the screenshot below) so that reviewers can access the preview.

image

File: articles/Document Information Extraction with Large Language Models/README.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
+---
+title: "Document Information Extraction with Large Language Models (framework, methods,  ensemble, and evaluation)"
+author: ["Carvin Li / ML Engineer", "Anjing Wang / NLP, ML Engineer"]
+createTime: 2024-05-21
+tags: ["Information Extraction", "Large Language Model"]
+thumb: ""
```
Comment (live9080) [inline articles/Document Information Extraction with Large Language Models/README.md:6]: Could you add the `thumb` and `thumb_h` and replace the images in the document  as the design? Thanks
https://www.figma.com/design/GHDhmnzRQTAUdjNTgOPJmF/57Blocks-Website-V3?node-id=14042-2730&t=XfpMSvxqOyvotTWG-0

Comment (Carvin618) [reply articles/Document Information Extraction with Large Language Models/README.md]: Sure~

File: articles/Document Information Extraction with Large Language Models/README.md [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+Complex situations can arise, particularly with range values. For instance, when production volume and capacity are uncertain, outputs may vary and be represented as a range in the text. Crafting prompts to uniformly capture and describe range values becomes crucial in such instances.
+
+Moreover, unit measurements may be required to specify in prompt. To make evaluation easy, we expect LLMs to present units in some firm format if the result comes with a unit. To standardize units through prompts, model outputs often exhibit inconsistencies. Therefore, additional processing such as unit convert may be required to ensure unit forms closely match those in the Ground Truth.
+
+### **Results Ensemble**
+
+Ensemble LLMs amalgamate predictions from multiple LLMs using diverse aggregation strategies like averaging, voting and stacking. This approach is motivated by addressing limitations inherent in individual LLMs, such as understanding complex contexts, performing mathematical calculations, or avoiding hallucination. Ensembles offer a promising solution by aggregating predictions, enhancing overall performance, and bolstering robustness across varied tasks and datasets.
```
Comment (wholehope) [inline articles/Document Information Extraction with Large Language Models/README.md:52]: diverse aggregation -> various

File: articles/Document Information Extraction with Large Language Models/README.md [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+During evaluation, we exclude properties marked as "not mentioned" in both the ground truth and LLM results from statistical analysis. Our focus lies on properties with values presented in the Ground Truth but differing from LLMs’ results.
+
+### **Confusion Matrix (Customized)**
+
+We defined a set of TP/TN/FP/FN to evaluate the results.
+
+![img](metrics_1.png)
+
+The below table is the confusion matrix using the definitions above.
```
Comment (wholehope) [inline articles/Document Information Extraction with Large Language Models/README.md:80]: You can remove the first confusion matrix and use only the second one.

File: articles/Document Information Extraction with Large Language Models/README.md [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+
+![img](https://lh7-us.googleusercontent.com/yVmgel3PqSGuc90UV6dim4UDGpESUGBjk6dy47RjDsztWg1fL31Eh1A_7hsyH81Mqfgj8Xjrq5Je_2neneetL1QdyVPsLbtrZxDl2euPyUv0Jyh-dI2RhYm9zPdYN5g4Rbtw4TJHhWbsdkF9Q16XIvc)
+
+There is another version of the confusion matrix that may be a little bit easier to understand.
+
```
Comment (wholehope) [inline articles/Document Information Extraction with Large Language Models/README.md:85]: There is a typo on this image.

File: articles/Document Information Extraction with Large Language Models/README.md [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+| $Precision$  | $TP/(TP+FP)$                                    | standard metric; when there is a POSITIVE extraction, the probability that this extraction is right. From a data consumer perspective, this shall be the most important metric.                                                                                                                                                                                                                 |
+| $F1$         | $2 \* recall \* precision / (recall+precision)$ | The standard metric to show the balance between recall and precision.                                                                                                                                                                                                                                                                                                                           |
+
+## 5. **Error Results Analysis**
```
Comment (wholehope) [inline articles/Document Information Extraction with Large Language Models/README.md:104]: Error Analysis

File: articles/Document Information Extraction with Large Language Models/README.md [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+
+To mitigate this issue, we minimize the reliance on LLMs for mathematical operations whenever possible. For example, when extracting the "Water-to-Oil Ratio", we may concurrently extract the necessary factors for calculation. If the extraction is successful, the result would be reserved. Otherwise, if the contributing factors are extracted, we calculate the ratio. If neither is extracted, we would assign a value of "not mentioned" to the property. Additionally, the contributing factors themselves may not always be included in the extraction result evaluation.
+
+### **Human Labeling Errors** 
+
+Unfortunately, the Ground Truth is not always accurate. Various errors may occur during the data annotation process, such as omissions (missing to label) and inaccuracies (incorrect labels). Strictly speaking, these human labeling errors should not be attributed to LLMs’ mistake but to inaccuracies in the Ground Truth. Therefore, when reviewing extraction results, we also need to cross-check the Ground Truth against the original text to enhance its accuracy.
```
Comment (wholehope) [inline articles/Document Information Extraction with Large Language Models/README.md:122]: You need to rephrase this paragraph. Ground truth conceptually is always 100% correct. The human annotation results used as ground truth but can be wrong.

File: articles/Document Information Extraction with Large Language Models/README.md [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+
+### **Design of Meta-Model of Ensemble**
+
+Designing a meta-model for Ensemble Stacking presents another challenge. Without Ground Truth reference, determining which model performs best for each property when ensemble different LLM results is very difficult. We've explored a meta-model design based on past evaluation results and LLM-provided confidence, marginally improving $TP\\%$ values by approximately $3\\%$ - $5\\%$. Further research may be needed for continued enhancement.
+
+### **Data Annotation**
+
+Data annotation for information extraction is notably challenging. Annotators, particularly for specialized articles, must be professionals or experts to ensure annotation’s accuracy and completeness. Reading articles is time-consuming, especially with thousands of documents. Initial annotation efforts began with a few papers, gradually increasing to dozens, and eventually hundreds or thousands as the program developed. However, this quantity still falls short of reflecting the overall distribution realistically. While programs can alleviate some initial annotation work, manual validation remains necessary, with the required time increasing alongside text volume.
```
Comment (wholehope) [inline articles/Document Information Extraction with Large Language Models/README.md:140]: specialized articles -> academic papers

File: articles/Document Information Extraction with Large Language Models/README.md (hunk 3 above)
Comment (Carvin618) [reply articles/Document Information Extraction with Large Language Models/README.md:80]: Sure~

File: articles/Document Information Extraction with Large Language Models/README.md [hunk 8]
```diff
@@ -0,0 +1,140 @@
+---
+title: "Document Information Extraction with Large Language Models (framework, methods,  ensemble, and evaluation)"
+author: ["Carvin Li / ML Engineer", "Anjing Wang / NLP, ML Engineer"]
+createTime: 2024-05-21
+tags: ["Information Extraction", "Large Language Model"]
+thumb: "thumb_h.png"
+thumb_b: "thumb_h.png"
```
Comment (live9080) [inline articles/Document Information Extraction with Large Language Models/README.md:7]: The `thumb_b` and `thumb` should have the different images which from the designer.  

One is a horizontal picture, one is a vertical picture

File: articles/Document Information Extraction with Large Language Models/thumb_h.png
Comment (live9080) [inline articles/Document Information Extraction with Large Language Models/thumb_h.png:1]: You can using the online tool to compress the image to smaller size.

https://tinypng.com/

File: articles/Document Information Extraction with Large Language Models/README.md (hunk 8 above)
Comment (Carvin618) [reply articles/Document Information Extraction with Large Language Models/README.md:7]: Does the horizontal image needs to keep the words direction on the image? Or simply rotate is ok?

File: articles/Document Information Extraction with Large Language Models/README.md (hunk 6 above)
Comment (Carvin618) [reply articles/Document Information Extraction with Large Language Models/README.md]: ok

File: articles/Document Information Extraction with Large Language Models/README.md (hunk 8 above)
Comment (Carvin618) [reply articles/Document Information Extraction with Large Language Models/README.md:7]: Done.

**Review by live9080 — APPROVED** [review]

**Pull request merged**